### ISSUE TITLE
Robustness to small overlap rate and complicated scenes

### ISSUE THREAD

**Issue by BoomSky0416** [nie-lang/UnsupervisedDeepImageStitching#17]
After I tried your pretrained network on my own data, It doesn't work to estimate a right homography. Does it mean this network can't work for low overlap rate or complicated scenes? Here is the data I use.
![ud_pipe_0](https://user-images.githubusercontent.com/57380777/152982135-9d2b8bea-f4ea-45bd-9118-0a2247960e4e.jpg)
![ud_pipe_1](https://user-images.githubusercontent.com/57380777/152982181-360881ac-4fb8-4f42-a345-80ae62e51e42.jpg)

**Comment by BoomSky0416** [comment]
Here is the result from this two stage network.
![image](https://user-images.githubusercontent.com/57380777/153104496-df1504ce-9ea9-4550-a84e-e491f8f38f49.png)

**Comment by BoomSky0416** [comment]
Here is the result from conventional feature-based image stitching method.
![image](https://user-images.githubusercontent.com/57380777/153105190-9382d301-63c0-4105-a6f0-df5ea2fd53f8.png)

**Comment by nie-lang** [comment]
Maybe it cannot work for cases with such a low overlap rate. 
1) In your case, the overlap rate is pretty low, which is different from our training dataset. Retraining/Finetuning our model in a new dataset (a dataset with a low overlap rate) may help it. 
2) In my view, a low overlap rate can be a big challenge for all the deep homography methods because of the pixel-level alignment loss. The lower the overlap rate is, the weaker the alignment constraint is.

**Comment by BoomSky0416** [comment]
Thanks for your explanation :).
